### PR TITLE
Feature/update packaging

### LINKED ISSUE #190: Support and deploy for Python 3.8
https://docs.python.org/3/whatsnew/3.8.html

> Python 3.8 was released on October 14th, 2019.

I think it would be important to have a release version that officially supports Python 3.8 and also deploy for it.

### PR DESCRIPTION
**Pull request by bdice**
## Description
This PR combines several changes that improve the packaging of signac-flow.

Goals:
- simplify deployment of releases of the package
- improve usage of the metadata fields in `setup.py`
- reduce the number of files in the repository root
- update to testing Python 3.8 and signac 1.3

If this is approved, I plan to adopt similar changes for the signac and signac-dashboard packages.

## Motivation and Context
I'm hoping to reduce the complexity of the release process, even if in just a few minor ways.

Resolves #190.

## Types of Changes

- [ ] Documentation update
- [ ] Bug fix
- [x] New feature
- [ ] Breaking change<sup>1</sup>

<sup>1</sup>The change breaks (or has the potential to break) existing functionality.

## Checklist:
- [x] I am familiar with the [**Contributing Guidelines**](https://github.com/glotzerlab/signac-flow/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [**Contributor Agreement**](https://github.com/glotzerlab/signac-flow/blob/master/ContributorAgreement.md).
- [x] My name is on the [list of contributors](https://github.com/glotzerlab/signac-flow/blob/master/contributors.yaml).
- [x] My code follows the [code style guideline](https://github.com/glotzerlab/signac-flow/blob/master/CONTRIBUTING.md#code-style) of this project.
- [x] The changes introduced by this pull request are covered by existing or newly introduced tests.

If necessary:
- [ ] I have updated the API documentation as part of the package doc-strings.
- [ ] I have created a separate pull request to update the [framework documentation](https://docs.signac.io/) on [signac-docs](https://github.com/glotzerlab/signac-docs) and linked it here.
- [x] I have updated the [changelog](https://github.com/glotzerlab/signac-flow/blob/master/changelog.txt).

### CODE REVIEW

**Comment by bdice** [comment]
@csadorf I explicitly requested your review on this PR since this PR adjusts some fields in `setup.py`. I added maintainer information and adapted the author information to match the example of numpy (one author with "et al." and generic team contact info for the maintainer): https://github.com/numpy/numpy/blob/master/setup.py#L392-L399

File: .circleci/config.yml [hunk 1]
```diff
@@ -71,18 +71,22 @@ jobs:
...
     <<: *test-template
     environment:
-      SIGNAC_VERSION: "v1.1.0"
-  test-3.7-signac-1.2.0:
+      SIGNAC_VERSION: "v1.2.0"
+  test-3.8-signac-1.3.0:
     <<: *test-template
     environment:
-      SIGNAC_VERSION: "v1.2.0"
+      SIGNAC_VERSION: "v1.3.0"
```
Comment (bdice) [inline .circleci/config.yml]: I decided to keep 1.0 and the two most recent minor versions -- we could test 1.1 if needed but I think that's overkill in our build matrix.

Comment (csadorf) [reply .circleci/config.yml]: Fair enough. Alternatively we test all minor versions only for release branches?

**Comment by tommy-waltmann** [comment]
Everything looks good to me.

**Pull request merged**